sdk/src/quotes/public/decrease-liquidity-quote.ts

```diff
@@ -91,115 +91,126 @@ export function decreaseLiquidityQuoteByLiquidity(
+  } = PriceMath.getSlippageBoundForSqrtPrice(params.sqrtPrice, params.slippageTolerance);
+
+  const { tokenEstA: tokenEstALower, tokenEstB: tokenEstBLower } = getTokenEstimatesFromLiquidity({
+    ...params,
+    sqrtPrice: sLowerSqrtPrice,
+    tickCurrentIndex: sLowerIndex,
+  });
 
-  const sqrtPriceX64 = sqrtPrice;
-  const sqrtPriceLowerX64 = PriceMath.tickIndexToSqrtPriceX64(tickLowerIndex);
-  const sqrtPriceUpperX64 = PriceMath.tickIndexToSqrtPriceX64(tickUpperIndex);
+  const { tokenEstA: tokenEstAUpper, tokenEstB: tokenEstBUpper } = getTokenEstimatesFromLiquidity({
+    ...params,
+    sqrtPrice: sUpperSqrtPrice,
+    tickCurrentIndex: sUpperIndex,
+  });
 
-  const tokenEstA = getTokenAFromLiquidity(liquidity, sqrtPriceX64, sqrtPriceUpperX64, false);
-  const tokenMinA = adjustForSlippage(tokenEstA, slippageTolerance, false);
-  const tokenEstB = getTokenBFromLiquidity(liquidity, sqrtPriceLowerX64, sqrtPriceX64, false);
-  const tokenMinB = adjustForSlippage(tokenEstB, slippageTolerance, false);
+  const tokenMinA = BN.min(BN.min(tokenEstA, tokenEstALower), tokenEstAUpper);
+  const tokenMinB = BN.min(BN.min(tokenEstB, tokenEstBLower), tokenEstBUpper);
 
   const tokenMinAExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenMinA, tokenExtensionCtx.tokenMintWithProgramA, tokenExtensionCtx.currentEpoch);
   const tokenEstAExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenEstA, tokenExtensionCtx.tokenMintWithProgramA, tokenExtensionCtx.currentEpoch);
   const tokenMinBExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenMinB, tokenExtensionCtx.tokenMintWithProgramB, tokenExtensionCtx.currentEpoch);
   const tokenEstBExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenEstB, tokenExtensionCtx.tokenMintWithProgramB, tokenExtensionCtx.currentEpoch);
 
   return {
-    liquidityAmount: liquidity,
-    tokenMinA: tokenMinAExcluded.amount,
-    tokenMinB: tokenMinBExcluded.amount,
-    tokenEstA: tokenEstAExcluded.amount,
-    tokenEstB: tokenEstBExcluded.amount,
+    tokenMinA,
+    tokenMinB,
+    tokenEstA,
+    tokenEstB,
+    liquidityAmount: params.liquidity,
     transferFee: {
       deductedFromTokenMinA: tokenMinAExcluded.fee,
       deductedFromTokenMinB: tokenMinBExcluded.fee,
       deductedFromTokenEstA: tokenEstAExcluded.fee,
       deductedFromTokenEstB: tokenEstBExcluded.fee,
     },
-  };
+  }
 }
 
-function quotePositionAboveRange(param: DecreaseLiquidityQuoteParam): DecreaseLiquidityQuote {
-  const { tickLowerIndex, tickUpperIndex, liquidity, slippageTolerance: slippageTolerance, tokenExtensionCtx } = param;
+function getTokenEstimatesFromLiquidity(params: DecreaseLiquidityQuoteParam) {
+  const { liquidity, tickLowerIndex, tickUpperIndex, sqrtPrice } = params;
 
-  const sqrtPriceLowerX64 = PriceMath.tickIndexToSqrtPriceX64(tickLowerIndex);
-  const sqrtPriceUpperX64 = PriceMath.tickIndexToSqrtPriceX64(tickUpperIndex);
+  if (liquidity.eq(ZERO)) {
```

Comment:
hmm, `decreaseLiquidityQuoteByLiquidityWithParamsUsingPriceSlippage` has early return path for zero liquidity.
On the other hand, original `decreaseLiquidityQuoteByLiquidityWithParams` doesn't.
So how about adding early return logic for it.
(or simply return zero here 🤔)